source/jormungandr/jormungandr/interfaces/v1/Journeys.py

```diff
@@ -488,11 +489,50 @@ def _set_specific_params(mod):
         # Add the interpreted parameters to the stats
         self._register_interpreted_parameters(args)
 
+        logger = logging.getLogger(__name__)
+
+        # generate an id that :
+        #  - depends on the coverage and differents arguments of the request (from, to, datetime, etc.)
+        #  - does not depends on the order of the arguments in the request (only on their values)
+        #  - does not depend on the _override_scenario argument
+        #  - if  _override_scenario is present, its type is added at the end of the id in
+        #    order to identify identical requests made with the different scenarios
+        #  - we add the current_datetime of the request so as to differentiate
+        #    the same request made at distinct moments
+        def generate_request_id():
+
+            path = str(flask.request.path)
+            args_for_id = dict(flask.request.args)
+            if "_override_scenario" in args_for_id:
+                scenario = str(args_for_id["_override_scenario"][0])
+                args_for_id["_override_scenario"] = ""
+            else:
+                scenario = "new_default"
+
+            json_repr = json.dumps(args_for_id, sort_keys=True, ensure_ascii=True)
+            # we could use the json_repr as an id, but we hash it to have something smaller
+            m = hashlib.sha256()
+            m.update(json_repr)
+            json_hash = m.hexdigest()
+
+            now = dt_to_str(datetime.datetime.utcnow())
+
+            result = "journeys_{}_{}#{}#".format(json_hash, now, scenario)
+
+            logger.info("Generating id : {} for request : {}".format(result, flask.request.url))
+
+            return result
+
+        request_id = generate_request_id()
+        args["request_id"] = request_id
+
         # If there are several possible regions to query:
         # copy base request arguments before setting region specific parameters
         if len(possible_regions) > 1:
             base_args = deepcopy(args)
 
+        logger.debug("We are about to ask journeys on regions : {}".format(possible_regions))
```

Comment:
There's a similar log here:
https://github.com/CanalTP/navitia/blob/ed9a2b2cccbc503b47d4d13c2a9005bba8e59678/source/jormungandr/jormungandr/interfaces/v1/Journeys.py#L400
Isn't it enough?

Reply:
well spotted ! Removed